### ISSUE TITLE
Spaces are stripped between consecutive tags

### ISSUE THREAD

**Issue by nickcernis** [nickcernis/html-to-markdown#9]
This test currently fails:

`$this->html_gives_markdown("<b>Bold</b> <i>Italic</i>", "**Test** *Italic*");`

Output:

    Expected :**Test** *Italic*
    Actual   :**Bold***Italic*

Need a better way to preserve spaces that exist between consecutive span tags.

**Comment by colinodell** [comment]
This issue seems to be caused by [line 78](https://github.com/nickcernis/html-to-markdown/blob/master/HTML_To_Markdown.php#L78):

~~~php
$html = preg_replace('~>\s+<~', '><', $html); // Strip white space between tags to prevent creation of empty #text nodes
~~~

This is stripping out the space which originally appeared between the two tags.

I did a quick test, and removing this line seemed to resolve the issue.  However, the lists aren't excluding those text nodes when determining their positions, so they go `1`, `3`, `5`, etc.  I'll need to somehow exclude those.

**Comment by colinodell** [comment]
Fixed by #28
